skrub/_matching.py

```diff
@@ -0,0 +1,119 @@
+import numpy as np
+from scipy import sparse
+from sklearn.base import BaseEstimator
+from sklearn.neighbors import NearestNeighbors
+
+
+class Matching(BaseEstimator):
+    def __init__(self):
+        pass
+
+    def fit(self, aux, main):
+        del main
+        self.aux_ = aux
+        self.neighbors_ = NearestNeighbors(n_neighbors=1).fit(aux)
+        return self
+
+    def match(self, main, max_dist):
+        distances, indices = self.neighbors_.kneighbors(main, return_distance=True)
+        distances, indices = distances.ravel(), indices.ravel()
+        reference_distances = self._get_reference_distances(main, indices, distances)
+        rescaled_distances = self._rescale_distances(distances, reference_distances)
+        return {
+            "index": indices,
+            "distance": distances,
+            "rescaled_distance": rescaled_distances,
+            "match_accepted": rescaled_distances <= max_dist,
+        }
+
+    def _get_reference_distances(self, main, indices, distances):
+        del main, indices, distances
+        return 1.0
+
+    def _rescale_distances(self, distances, reference_distances):
+        reference_distances = np.asarray(reference_distances)
+        ref_is_zero = reference_distances == 0.0
+        rescaled_distances = np.zeros_like(distances)
+        rescaled_distances[~ref_is_zero] = (
+            distances[~ref_is_zero] / reference_distances[~ref_is_zero]
+        )
+        rescaled_distances[ref_is_zero] = np.inf
+        rescaled_distances[distances == 0] = 0.0
+        return rescaled_distances
+
+
+def _sample_pairs(n, n_pairs, random_state):
+    assert n > 1
+    assert n_pairs > 0
+    rng = np.random.default_rng(random_state)
```

Comment:
I used check_random_state to make it more familiar but I made the default 0 (ie a seeded generator) rather than the usual None

not very important but check_random_state returns a RandomState whereas numpy now recommends using the Generator instead